clients/samples/gemm_tuning/sample_hipblaslt_gemm_tuning_splitk_ext.cpp

```diff
@@ -0,0 +1,170 @@
+                               void*              d_b,
+                               void*              d_c,
+                               void*              d_d,
+                               void*              d_workspace,
+                               int64_t            max_workspace_size,
+                               hipStream_t        stream)
+{
+    // Get all algo doesn't require a gemm instance.
+    std::vector<hipblasLtMatmulHeuristicResult_t> heuristicResult;
+    CHECK_HIPBLASLT_ERROR(hipblaslt_ext::getAllAlgos(handle,
+                                                     hipblaslt_ext::GemmType::HIPBLASLT_GEMM,
+                                                     trans_a,
+                                                     trans_a,
+                                                     HIPBLASLT_R_16F,
+                                                     HIPBLASLT_R_16F,
+                                                     HIPBLASLT_R_16F,
+                                                     HIPBLASLT_R_16F,
+                                                     HIPBLASLT_COMPUTE_F32,
+                                                     heuristicResult));
+
+    hipblaslt_ext::GemmPreference gemmPref;
+    gemmPref.setMaxWorkspaceBytes(max_workspace_size);
+    hipblaslt_ext::Gemm gemm(handle,
+                             trans_a,
+                             trans_b,
+                             HIPBLASLT_R_16F,
+                             HIPBLASLT_R_16F,
+                             HIPBLASLT_R_16F,
+                             HIPBLASLT_R_16F,
+                             HIPBLASLT_COMPUTE_F32);
+
+    hipblaslt_ext::GemmEpilogue
+        epilogue; // No action needed, default is HIPBLASLT_EPILOGUE_DEFAULT. (Gemm only)
+    hipblaslt_ext::GemmInputs inputs;
+    inputs.a     = d_a;
+    inputs.b     = d_b;
+    inputs.c     = d_c;
+    inputs.d     = d_d;
+    inputs.alpha = &alpha;
+    inputs.beta  = &beta;
+    gemm.setProblem(m, n, k, batch_count, epilogue, inputs);
+
+    hipblaslt_ext::GemmTuning tuning;
+    tuning.splitK = 8;
+
+    uint64_t            workspace_size = 0;
+    std::vector<size_t> validIdx;
+    for(size_t i = 0; i < heuristicResult.size(); i++)
+    {
+        size_t workspaceSizeInBytes = 0;
+        // If tuning is given, the API will not return success if the solution cannot
+        // accept an user tuning parameter.
+        if(gemm.isAlgoSupported(heuristicResult[i].algo, tuning, workspaceSizeInBytes)
+           == HIPBLAS_STATUS_SUCCESS)
+        {
+            if(workspaceSizeInBytes <= max_workspace_size)
+            {
+                workspace_size = max(workspace_size, workspaceSizeInBytes);
+                validIdx.push_back(i);
+            }
+        }
+    }
+
+    if(validIdx.empty())
+    {
+        std::cerr << "No valid solution found!" << std::endl;
+        return;
+    }
+
+    // In this sample, the workspace is already allocated with max_workspace_size
```

Comment:
if requried workspace size is less than ma_workspace_size, we should allocate it here.
There example is for GSU which is workspace sensitive feature. we should let user know it.

Reply:
fixed